lib/elixir/lib/system.ex

```diff
@@ -637,6 +637,37 @@ defmodule System do
     end
   end
 
+  @doc """
+  Similar to `System.cmd/3` but raises an error on non-zero exit status.
+
+  ## Examples
+
+      iex> System.cmd! "echo", ["hello"]
+      "hello\n"
+
+  """
+  @spec cmd!(binary, [binary], keyword) :: Collectable.t()
+  def cmd!(command, args, opts \\ []) when is_binary(command) and is_list(args) do
+    case cmd(command, args, opts) do
+      {result, 0} ->
+        result
+
+      {"", status} ->
+        raise "command exited with status #{inspect(status)}"
+
+      {result, status} when is_binary(result) ->
+        result =
+          result
+          |> String.split("\n")
+          |> Enum.map_join("\n", &("        " <> &1))
```

Comment:
I usually use String.replace in cases like this: `|> String.replace("\n", "        \n")`.